Refactor-Remote-Data-Source-to-Retrofit-OkHttp (movie and series)

Mostafa-khaled21 opened the pull request:
Migrated both Movies and Series remote data sources from Ktor to Retrofit.
Fixed issue where release date was not displaying properly in Series UI .

https://github.com/user-attachments/assets/d8bf999e-6061-4e16-a3b1-683cf5cb1687

Mostafa-khaled21 committed:
Switch Series module from Ktor to Retrofit

Mostafa-khaled21 committed:
Implement Movie Retrofit Remote Data Source

Mostafa-khaled21 committed:
fix :Add date property to SeriesUIModel
-it does not appear in the design in develop

skaik-mo commented on app/src/main/java/com/giraffe/cineverseapp/di/networkRetrofitModule.kt:
```diff
@@ -0,0 +1,74 @@
+
+val networkRetrofitModule = module {
+
+    single(named(BASE_URL)) { BuildConfig.BASE_URL }
```
`BASE_URL` and `ACCESS_TOKEN` are duplicated in networkModule.

Mostafa-khaled21 replied:
networkModule for ktor.after using retrofit completely it will be stopped

skaik-mo commented on datasource/remote/media/src/main/java/com/giraffe/media/movie/MoviesApiServiceRetrofit.kt:
```diff
@@ -0,0 +1,42 @@
+import retrofit2.http.*
+
+interface MoviesApiServiceRetrofit {
+    @GET("movie/{id}")
```
You can enhance the endpoint by defining it as a constant in a companion object and reusing it, as we’ve done in other places.

skaik-mo replied:
Take a look at the `MoviesRemoteDataSourceImp` file

skaik-mo commented on datasource/remote/media/src/main/java/com/giraffe/media/series/SeriesApiServiceRetrofit.kt:
```diff
@@ -0,0 +1,44 @@
+
+interface SeriesApiServiceRetrofit {
+
+    @GET("search/tv")
```
Also here, consider extracting the endpoint into a `companion object `constant for reuse

skaik-mo commented on datasource/remote/media/src/main/java/com/giraffe/media/movie/response/MoviesListResponse.kt:
```diff
@@ -4,7 +4,6 @@ import  com.giraffe.media.movie.datasource.remote.dto.MovieDto
 import kotlinx.serialization.SerialName
 import kotlinx.serialization.Serializable
```
Remove unused import for cleaner code.

skaik-mo commented on datasource/remote/media/src/main/java/com/giraffe/media/series/response/SeriesResponse.kt:
```diff
@@ -1,15 +1,14 @@
 import com.giraffe.media.series.datasource.remote.dto.SeriesDto
-import kotlinx.serialization.SerialName
-import kotlinx.serialization.Serializable
+import com.google.gson.annotations.SerializedName
```
Why is `kotlinx.serialization.SerialName` used in `MoviesListResponse`, but a different one here? Let’s keep the import consistent.

skaik-mo reviewed (changes requested):
You can create a `retrofit` package inside each feature to organize related classes. For example, the `retrofit` package in the movies feature could include `MoviesApiServiceRetrofit` and `MoviesRemoteDataSourceImplRetrofit`. Do the same for other features like series and person, it keeps the structure modular and easier to navigate.

Mostafa-khaled21 committed:
Refactor: Move Retrofit API services and data sources to retrofit package

Mostafa-khaled21 committed:
Refactor: Relocate Retrofit API service interfaces

- Move `MoviesApiServiceRetrofit` to `com.giraffe.media.movie.retrofit` package.
- Move `SeriesApiServiceRetrofit` to `com.giraffe.media.series.retrofit` package.
- Update `networkRetrofitModule.kt` to reflect new package locations.

Mostafa-khaled21 committed:
Merge remote-tracking branch 'origin/develop' into feature/Refactor-Remote-Data-Source-to-Retrofit-OkHttp

# Conflicts:
#	gradle/libs.versions.toml

Mostafa-khaled21 committed:
Add retrofit and okhttp to version catalog

These libraries are added to the `libs.versions.toml` file to manage their versions centrally.

Ashraf-El-Fallah commented on repository/media/src/main/java/com/giraffe/media/movie/datasource/remote/dto/ReviewsResponseDto.kt:
```diff
@@ -1,32 +1,44 @@
 
-@Serializable
 data class ReviewsResponseDto(
+    @SerializedName("id")
```
I don't think used SerializedName annotation here is necessary because it's the same name of the variable .. also in other variables like content and results

skaik-mo committed:
Merge branch 'develop' into feature/Refactor-Remote-Data-Source-to-Retrofit-OkHttp

Mostafa-khaled21 committed:
Refactor: Remove redundant @SerializedName annotations from DTOs
